src/Timer.h

```diff
@@ -53,17 +53,11 @@ template<typename T>
 void
 Timer::run(const std::chrono::milliseconds interval, T callback)
 {
-    auto referenceTime = std::chrono::steady_clock::now();
+    auto wakeupTime = std::chrono::steady_clock::now();
     while(true) {
-        using namespace std::chrono_literals;
-        const auto elapsedTime = std::chrono::steady_clock::now() - referenceTime;
-        const auto diff = interval - elapsedTime;
-        if(diff > 0ms) {
-            std::this_thread::sleep_for(diff);
-        }
+        wakeupTime = wakeupTime + interval;
+        std::this_thread::sleep_until(wakeupTime);
```

Comment:
Is sleep_until free of spurious wakeups?

Reply:
https://stackoverflow.com/questions/30538508/can-stdthis-threadsleep-for-have-spurious-wakeups

Reply:
According to the standard - yes.
The discussion is about bug in gcc 5

Reply:
The `std::steady_clock` is used:
>Class std::chrono::steady_clock represents a monotonic clock. The time points of this clock cannot decrease as physical time moves forward and the time between ticks of this clock is constant. This clock is not related to wall clock time (for example, it can be time since last reboot), and is most suitable for measuring intervals.
https://en.cppreference.com/w/cpp/chrono/steady_clock